cranelift-frontend/src/switch.rs

```diff
@@ -129,21 +169,38 @@ impl Switch {
     ) {
         for ContiguousCaseRange { first_index, ebbs } in contiguous_case_ranges.into_iter().rev() {
             if ebbs.len() == 1 {
-                let is_good_val = bx.ins().icmp_imm(IntCC::Equal, val, first_index as i64);
-                bx.ins().brnz(is_good_val, ebbs[0], &[]);
+                if first_index == 0 {
+                    bx.ins().brz(val, ebbs[0], &[]);
+                } else {
+                    let is_good_val = bx.ins().icmp_imm(IntCC::Equal, val, first_index as i64);
+                    bx.ins().brnz(is_good_val, ebbs[0], &[]);
+                }
             } else {
                 let jt_ebb = bx.create_ebb();
-                let is_good_val =
-                    bx.ins()
-                        .icmp_imm(IntCC::UnsignedGreaterThanOrEqual, val, first_index as i64);
-                bx.ins().brnz(is_good_val, jt_ebb, &[]);
-                cases_and_jt_ebbs.push((first_index, jt_ebb, ebbs));
+                if first_index == 0 {
+                    // if first_index is 0, then icmp_imm uge val, first_index is trivially true
```

Comment:
nit: can you put backquotes around the cranelift instruction, and parenthesis within it, to make it more readable? (and add a dot at the end)

Reply:
> and parenthesis within it

Where?

Reply:
I thought around the instructions, but the backquotes make it clear enough already, so this is fine as is. Thanks.